loopin.go

```diff
@@ -840,6 +876,26 @@ func (s *loopInSwap) waitForSwapComplete(ctx context.Context,
 	htlcKeyRevealed := false
 	for !htlcSpend || !invoiceFinalized {
 		select {
+		// If the client requested the swap to be abandoned, we override
+		// the status in the database.
+		case <-s.abandonChan:
+			err = s.setStateAbandoned(ctx)
```

Comment:
Cannot really happen here but just for future proofing, if the state isn't pending (or considered abandonable) then we'd still cancel the invoice etc here.

Reply:
Nice catch!

Reply:
see https://github.com/lightninglabs/loop/pull/661#discussion_r1403395076